org.opentosca.container.engine.plan.plugin.bpel/src/org/opentosca/container/engine/plan/plugin/bpel/BpelPlanEnginePlugin.java

```diff
@@ -140,6 +141,155 @@ public String getLanguageUsed() {
+            }
+        }
+        catch (final TransformerConfigurationException e) {
+            BpelPlanEnginePlugin.LOG.error("Couldn't load BPELRESTLightUpdater", e);
+        }
+        catch (final ParserConfigurationException e) {
+            BpelPlanEnginePlugin.LOG.error("Couldn't load BPELRESTLightUpdater", e);
+        }
+        catch (final SAXException e) {
+            BpelPlanEnginePlugin.LOG.error("ParseError: Couldn't parse .bpel file", e);
+        }
+        catch (final IOException e) {
+            BpelPlanEnginePlugin.LOG.error("IOError: Couldn't access .bpel file", e);
+        }
+
+        // package process
+        BpelPlanEnginePlugin.LOG.info("Prepare deployment of PlanModelReference");
+
+        if (this.fileAccessService != null) {
+            try {
+                if (tempPlan.createNewFile()) {
+                    // package the updated files
+                    BpelPlanEnginePlugin.LOG.debug("Packaging plan to {} ", tempPlan.getAbsolutePath());
+                    tempPlan = this.fileAccessService.zip(tempDir, tempPlan);
+                } else {
+                    BpelPlanEnginePlugin.LOG.error("Can't package temporary plan for deployment");
+                    return false;
+                }
+            }
+            catch (final IOException e) {
+                BpelPlanEnginePlugin.LOG.error("Can't package temporary plan for deployment", e);
+                return false;
+            }
+        }
+
+        // deploy process
+        BpelPlanEnginePlugin.LOG.info("Deploying Plan: {}", tempPlan.getName());
+        String processId = "";
+        Map<String, URI> endpoints = Collections.emptyMap();
+        try {
+            if (ENGINE.equalsIgnoreCase(BPS_ENGINE)) {
+                final BpsConnector connector = new BpsConnector();
+
+                processId = connector.deploy(tempPlan, URL, USERNAME, PASSWORD);
+
+                endpoints = connector.getEndpointsForPID(processId, URL, USERNAME, PASSWORD);
+            } else {
+                final OdeConnector connector = new OdeConnector();
+
+                processId = connector.deploy(tempPlan, URL);
+
+                endpoints = connector.getEndpointsForPID(processId, URL);
+            }
+        }
+        catch (final Exception e) {
+            e.printStackTrace();
+        }
+
+        // this will be the endpoint the container can use to instantiate the
+        // BPEL Process
+        URI endpoint = null;
+        if (endpoints.keySet().size() == 1) {
+            endpoint = (URI) endpoints.values().toArray()[0];
+        } else {
+            for (final String partnerLink : endpoints.keySet()) {
+                if (partnerLink.equals("client")) {
+                    endpoint = endpoints.get(partnerLink);
+                }
+            }
+        }
+
+        if (endpoint == null) {
+           
+            return false;
+        }
+
+        if (processId != null && endpoint != null && portType != null) {
```

Comment:
condidtion could be inverted to save a level of indentation for the current if-block, since the else-branch always returns.

Reply:
Changed